### PR TITLE
817 featureadmin add donated amount column and link to donations to campaign grid

### PR DESCRIPTION
**Pull request by borislavstoychev**
## Screenshots:

| Before              | After              |
| ------------------- | ------------------ |
| image|image|

 | After              |
|--------------------|
|image|

## Testing

**New environment variables**:

- `env var` : env var details

**New dependencies**:

- `dependency` : dependency details

**New dev dependencies**:

- `dependency` : dependency details

### CODE REVIEW

**Comment by borislavstoychev** [comment]
[changes in APi](https://github.com/podkrepi-bg/api/pull/262)

File: src/components/campaigns/grid/CampaignGrid.tsx [hunk 1]
```diff
@@ -119,53 +129,97 @@ export default function CampaignGrid() {
...
     },
     {
       field: 'updatedAt',
       headerName: t('campaigns:updatedAt'),
       align: 'left',
       width: 230,
       headerAlign: 'left',
+      renderCell: (cellValues: GridRenderCellParams) => (
+        <>
+          {formatRelative(parseISO(cellValues.row.updatedAt), new Date(), {
```
Comment (kachar) [inline src/components/campaigns/grid/CampaignGrid.tsx:218]: I think you can use `getRelativeDate` as well. It wraps the format relative with a `new date`

You may pass the locale as second param to `getRelativeDate` after adding it as optional to the fn

**Review by kachar — APPROVED** [review]
Well done, straight to the point :dart:

File: src/components/donations/grid/Grid.tsx [hunk 2]
```diff
@@ -24,10 +25,12 @@ interface PersonCellProps {
...
   const { t } = useTranslation()
-
-  const { data }: UseQueryResult<DonationResponse[]> = useDonationsList()
+  const router = useRouter()
   const { isDetailsOpen } = ModalStore
-
+  const id = router.query.id
+  const { data }: UseQueryResult<DonationResponse[]> = id
+    ? useCampaingDonationsList(id as string)
+    : useDonationsList()
```
Comment (igoychev) [inline src/components/donations/grid/Grid.tsx:33]: the id here is the campaignId, so please rename to campaignId for the code to be more readable

File: src/gql/campaigns.ts [hunk 3]
```diff
@@ -54,6 +54,7 @@ export type AdminCampaignResponse = BaseCampaignResponse & {
   coordinator: {
     person: { firstName: string; lastName: string }
   }
+  reachedAmound: number
```
Comment (igoychev) [inline src/gql/campaigns.ts:57]: typo: Amound -> reachedAmount

File: src/pages/admin/donations/campaign/[id].tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import DonationsPage from 'components/donations/DonationsPage'
+import { securedAdminProps } from 'middleware/auth/securedProps'
+import { endpoints } from 'service/apiEndpoints'
+
+export const getServerSideProps = securedAdminProps(
+  ['common', 'auth', 'admin', 'donations', 'validation'],
+  (ctx) => endpoints.campaign.getDonations(ctx.query.id as string).url,
+)
+
+export default DonationsPage
```
Comment (igoychev) [inline src/pages/admin/donations/campaign/[id].tsx:10]: Is there any reason for defining a new page or we can use the existing DonationPage and just filter the Donations with query param?
The initial idea was the campaign to come through a query param in this way: /admin/donations?campaignId=[uuid] while this implementation is creating a new route admin/donations/campaign/[id]

File: src/components/campaigns/grid/CampaignGrid.tsx [hunk 5]
```diff
@@ -119,53 +129,97 @@ export default function CampaignGrid() {
...
       align: 'left',
       width: 230,
       headerAlign: 'left',
+      renderCell: (cellValues: GridRenderCellParams) => (
+        <>
+          {formatRelative(parseISO(cellValues.row.startDate), new Date(), {
+            locale: i18n.language == 'bg' ? bg : enUS,
+          })}
+        </>
+      ),
```
[the remark this answers is not in the document]
Comment (borislavstoychev) [reply src/components/campaigns/grid/CampaignGrid.tsx]: image

**Comment by borislavstoychev** [comment]
image
@kachar  @igoychev  is it good idea to have grid cell expand, on mouse over? For example description field is 500 character and it is not possible to display all content.

File: src/gql/campaigns.ts [hunk 6]
```diff
@@ -54,6 +54,9 @@ export type AdminCampaignResponse = BaseCampaignResponse & {
   coordinator: {
     person: { firstName: string; lastName: string }
   }
+  incomingTransfers: [{ amount: number }]
+  outgoingTransfers: [{ amount: number }]
```
Comment (kachar) [inline src/gql/campaigns.ts:58]: Most probably this needs to be:

```suggestion
  incomingTransfers: { amount: number }[]
  outgoingTransfers: { amount: number }[]
```

In the campaign entity we can see:

https://github.com/podkrepi-bg/api/blob/90cb4c5e1a54208f2be181fc6088e846a2ff45e1/apps/api/src/domain/generated/campaign/entities/campaign.entity.ts#L36-L37

**Pull request merged**